src/Tempest/Framework/Testing/Http/TestResponseHelper.php

```diff
@@ -219,6 +219,78 @@ public function assertNotSee(string $search): self
         return $this;
     }
 
+    public function assertViewData(string $key, ?Closure $test = null): self
+    {
+        $data = $this->body->data;
+        $value = $data[$key];
+
+        Assert::assertArrayHasKey(
+            key: $key,
+            array: $data,
+            message: sprintf(
+                'No view data was set for [%s], available view data keys: %s',
+                $key,
+                implode(', ', array_keys($data)),
+            ),
+        );
+
+        if ($test !== null) {
+            $test($data, $value);
+        }
+
+        return $this;
+    }
+
+    public function assertViewDataMissing(string $key): self
+    {
+        $data = $this->body->data;
+
+        Assert::assertArrayNotHasKey(
+            key: $key,
+            array: $data,
+            message: sprintf('Failed asserting that view data key [%s] was not set', $key),
+        );
+
+        return $this;
+    }
+
+    public function assertViewDataAll(Closure $test): self
```

Comment:
Can you add docblocks to the closure so users can know what arguments it takes?

```suggestion
    public function assertViewDataAll(Closure $callback): self
```